### ISSUE TITLE
'Timeout getting IP address' error on healthcheck and ipsec containers in v1.2.0

### ISSUE THREAD

**Issue by samattridge** [rancher/rancher#6858]
**Rancher Version: 1.2.0**

**Docker Version: 1.12.1**

**OS and where are the hosts located? (cloud, bare metal, etc): cloud**

**Setup Details: single node rancher, external DB**

**Environment Type: Cattle**

Upgraded to Rancher 1.2.0 today after seeing it was in general release.  One server has upgraded perfectly well but a second server is continually struggling to startup a 'healthcheck' container as well as an 'ipsec' container.  It also additionally fails to startup a customized 'logrotate' container that we use.  All three failing containers result in the same error:

> Timeout getting IP address

Nothing particularly useful coming from the rancher-server logs:

> 2016-12-01 16:32:44,719 ERROR [7949843f-82d7-486d-a347-97224a8d0415:1411369] [instance:66195->instanceHostMap:29323] [instance.start->(InstanceStart)->instancehostmap.activate] [] [cutorService-29] [c.p.e.p.i.DefaultProcessInstanceImpl] Agent error for [compute.instance.activate.reply;agent=854]: Timeout getting IP address
2016-12-01 16:32:44,719 ERROR [7949843f-82d7-486d-a347-97224a8d0415:1411369] [instance:66195] [instance.start->(InstanceStart)] [] [cutorService-29] [i.c.p.process.instance.InstanceStart] Failed [1/2] to Starting for instance [66195]
2016-12-01 16:32:44,784 ERROR [7949843f-82d7-486d-a347-97224a8d0415:1411369] [instance:66195->instanceHostMap:29323] [instance.start->(InstanceStart)->instancehostmap.activate] [] [cutorService-29] [c.p.e.p.i.DefaultProcessInstanceImpl] Agent error for [compute.instance.activate.reply;agent=854]: Timeout getting IP address
2016-12-01 16:32:44,784 ERROR [7949843f-82d7-486d-a347-97224a8d0415:1411369] [instance:66195] [instance.start->(InstanceStart)] [] [cutorService-29] [i.c.p.process.instance.InstanceStart] Failed [2/2] to Starting for instance [66195]
2016-12-01 16:32:46,323 ERROR [:] [] [] [] [cutorService-15] [o.a.c.m.context.NoExceptionRunnable ] Expected state running but got stopped: Timeout getting IP address

Could you please advise on anything we can do to fix the issue or further diagnose?

Many thanks in advance.

**NOTE: FOR THOSE RUNNING BOOT2DOCKER**
If you are running network-services stack with plugin-manager:v0.2.12, please upgrade to the newest version of network services, there will be an "Upgrade Available" button to upgrade to network-manager:v0.2.13.

**NOTE: v1.2.0 only works with the docker bridge as `docker0` https://github.com/rancher/rancher/issues/6896 and docker must be installed in  `/var/lib/docker` https://github.com/rancher/rancher/issues/6897**
We will be making it more flexible in the next release. 

**NOTE: If neither of the above solutions fix your issue**

Can you please run the following commands with the CLI and share the files?

```
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
```

Please be advised the files will have all your hosts and IP in them.

**Comment by cjellick** [comment]
How many hosts do you have in the environement? Just those 2?

**Comment by jslatten** [comment]
I am experiencing the same issue. I can not get the ipsec or scheduler container to start properly. I keep seeing the Timeout error mentioned above. 

I am using boot2docker hosts with docker 1.12.1 

Has anyone actually been able to get Rancher 1.2.0 to work correctly? I have re-installed everything from scratch and rebuilt my server with new database, removed all containers from hosts, etc...Basically a clean slate.

**Comment by lucacri** [comment]
Same issue for me. It's happening on multiple servers and it brought down two of my sites. A general release should not bring downtime like this..

**Comment by jtcressy** [comment]
Exact same issues here, going to recreate all boot2docker hosts and redeploy containers since i have external NFS for persistence. My entire production platform is down right now, plus since i restarted all the boot2docker hosts it created duplicates for all 8 of my hosts. So now things are dancing all over the place starting/failing/starting/failing repeatedly. What a great "general availability" release here. Will report back with findings.

**Comment by adamgoose** [comment]
When I first upgraded a brand-new v1.1.4 single-node 2-host Rancher environment to v1.2.0, everything went mostly smoothly. All infra and user services were green.

Overnight, it all went to hell and started experiencing these issues. The following steps has led to all-green again (for now):

1. Using the Rancher UI, stop all services (just so Rancher stops trying to start them)
2. Stop Rancher Server with `docker stop rancher-server` or equivalent
3. On all hosts: `docker rm -f {every_container_id}`, including running and stopped containers
4. Start Rancher Server
5. Infra -> Hosts -> Add Host -> Custom -> Copy Command, and run it on all hosts
6. Wait for infra services to turn green
7. Slowly start user services

Will report back if things go south over time.

**Comment by pschirch** [comment]
I prepared my migration path in a clean docker machine setup. I created one machine for the rancher-server (1.1.4) and another one for a simple demo stack with nginx:stable and HA Proxy.

* rancher-server (Boot2Docker version 1.12.3)
* rancher-node-01 (Boot2Docker version 1.12.3)

The infrastructure stacks **healthcheck**, **ipsec** and **scheduler** ends up with `Expected state running but got error: Timeout getting IP address` and endless `Need to restart service reconcile`.

Thats really frustrating for a stable release.

**Comment by kallqvist** [comment]
+1
Same situation here; all my environments are currently down since the upgrade also redeployed all my loadbalancers and this issue prevented them from ever coming online again!

Healthcheck and scheduler stacks are alternating between 'timeout getting IP address' and 'need to restart service reconcile'.
I'm down to all except one host agent deactivated, removing and re-installing the remaining agent doesn't seem to bring any difference.

Sometimes I can also read a NullPointerException in the rancher master logs; can't tell if that is related or not?

`2016-12-01 20:22:40,643 ERROR [4cf7d31a-b187-481e-b056-b05ea2898b67:2190478] [service:410] [service.finishupgrade] [] [cutorService-11] [c.p.e.p.i.DefaultProcessInstanceImpl] Unknown exception java.lang.NullPointerException: null
	at io.cattle.platform.servicediscovery.process.ServiceFinishUpgrade$1.run(ServiceFinishUpgrade.java:39) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.activity.ActivityService.run(ActivityService.java:43) ~[cattle-activity-log-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.process.ServiceFinishUpgrade.handle(ServiceFinishUpgrade.java:33) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runHandler(DefaultProcessInstanceImpl.java:448) [cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:399) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$4.execute(DefaultProcessInstanceImpl.java:393) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.engine.idempotent.Idempotent.execute(Idempotent.java:42) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]`

**Comment by pschirch** [comment]
@jslatten No. I made a clean Rancher 1.2.0 install from scratch too, same issues.

**Comment by tpagden** [comment]
I'd also like to report the same thing - I'm using Vagrant with multiple instances. Timeout on the IP address.

Completely clean install with new VMs, no legacy whatsoever, and just nominal packages installed. The agent containers (healthcheck, scheduler, etc.) are bouncing and restarting.

**Comment by alena1108** [comment]
@kallqvist the exception you are seeing, is unrelated to ip address obtain failure

**Comment by kallqvist** [comment]
@alena1108 thanks! I removed that part from my comment to avoid creating confusion in here

**Comment by ibuildthecloud** [comment]
It appears b2d does not work.  To confirm that this is your issue, run `ln -s / /mnt/sda1` in the `network-manager` container.  After a bit the ipsec container will come up.  We will work on a fix for this.

**Comment by ibuildthecloud** [comment]
For anybody else not running b2d, can you please run the following commands with the CLI and share the files.

```bash
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
```

Please be advised the files will have all your hosts and IP in them.

**Comment by pschirch** [comment]
@ibuildthecloud `ln -s / /mnt/sda1` works! All infrastructure stacks comes online now.

**Comment by lucacri** [comment]
I just recreated the full environment (120 hosts) and rancher is still crapping out on the infrastructure services (ipsec, network-services, scheduler) with a ton of errors like this:

`2016-12-01 22:32:32,978 ERROR [902d8b92-a493-4a32-b12c-6ff1de043257:4900] [instance:346] [instance.start->(InstanceStart)] [] [cutorService-18] [i.c.p.process.instance.InstanceStart] Failed to Waiting for deployment unit instances to create for instance [346]`

As of now, both my sites have been down for 7 hours, and clients are mad. This is NOT what a stable upgrade should look like. After I finish fixing this mess, I'll look for more stable and serious alternatives

**Comment by ibuildthecloud** [comment]
@lucacri can you run the commands I shared.  You can email the files directly to me darren@rancher.com if you don't want to share with the world.

**Comment by lucacri** [comment]
@ibuildthecloud I sent you an email

**Comment by kallqvist** [comment]
@ibuildthecloud

ipsec log turned out empty.

metadata is pretty much only logging this over and over again:
```
network-services-metadata-1     | time="2016-12-01T22:35:12Z" level=info msg="Downloaded in 543.187019ms"
network-services-metadata-1     | time="2016-12-01T22:35:12Z" level=info msg="Loading answers"
network-services-metadata-1     | time="2016-12-01T22:35:12Z" level=info msg="Loaded answers"
network-services-metadata-1     | time="2016-12-01T22:35:12Z" level=info msg="Applied https://ranchermasters.ops.example.com/v1/configcontent/metadata-answers?version=2649-eb849cc452eab2e95f5d30ac7decee95"
```

network-manager however is logging a bunch of errors
mainly about problems reading resolv.conf and
`nsenter: cannot open /proc/6899/ns/ipc: No such file or directory`

[network-manager.txt](https://github.com/rancher/rancher/files/625952/network-manager.txt)

**Comment by samattridge** [comment]
@cjellick yes, just two hosts as this is my test system prior to rolling out to our other systems.

Checking again this morning, we are still seeing the same issue.  One server working, one not.  They are both Azure D1_v2 servers and registered in exactly the same manner.  I've also tried removing the host from Rancher and re-registering as well as rebuilding from scratch but still no dice.

**Comment by willseward** [comment]
I can reliably repro this on a b2d cluster. Is b2d supported for local testing?

**Comment by cjellick** [comment]
@samattridge can you follow the instructions that @ibuildthecloud for grabbing all the relevant logs and either posting them or emailing them over?

**Comment by iskrid** [comment]
Same issue here, completely new setup: Ubuntu16.04 docker 1.12.1, running a agent on same host and it wont start anything, just puking the "Expected state running but got error: instance is not running : Dependencies readiness error" message

**Comment by DomiStyle** [comment]
I'm also getting the error "Timeout getting IP address. Using 2 fresh Debian 8 VMs. Sent a mail to @ibuildthecloud with the log files.

**Comment by pschirch** [comment]
Tested it again with Vagrant, VirtualBox and a Ubuntu 16.04 node and Docker 1.12.3.. All infrastructure stacks works fine.

**Comment by tbohnen** [comment]
same issue, ipsec log that might help:
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=debug msg="About to start server and listen on port: :8111"
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=debug msg="Reloading ..."
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=info msg="Listening for ARP requests on eth0"
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=info msg="Listening on :8111"
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=error msg="couldn't get self container from metadata: Error 404 accessing /self/container path"
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=error msg="couldn't reload the overlay: Error 404 accessing /self/container path"
12/2/2016 10:23:22 PMtime="2016-12-02T20:23:22Z" level=fatal msg="Error 404 accessing /self/container path"
12/2/2016 10:23:22 PM00[LIB] loading feature SIGNER:HMAC_SHA2_512_256 in plugi

**Comment by pschirch** [comment]
@tbohnen the ln fix only solves the issue on boot2Docker

**Comment by pschirch** [comment]
@tbohnen while Ubuntu 16.04 with all available Updates and Docker 1.12.3 works for me, i'm interessted in your kernel and Docker version.

**Comment by stefanlasiewski** [comment]
I ran into this also. In my case, I had duplicate containers for the rancher/dns and rancher/metadata containers. See https://snag.gy/Uxr24e.jpg .

Some containers were named like `network-services_metadata_dns_1` (with underscores), which were created by Rancher v1.1.4. Containers named like `network-services-metadata-dns_1`, with  dashes in their name but not underscores, were created by v1.2.0.

Once I deleted the containers created by Rancher v1.1.4 I was able to get Rancher working again. I also had to run `docker ps -a |grep rancher | grep -v rancher/agent | while read CID junk; do docker rm $CID; done` on each host, and rancher/agent created all of the Infrastructure containers again, and most of those images worked.

**Comment by deniseschannon** [comment]
Anyone using boot2docker hosts must be running network services v0.0.2, which is running rancher/network-manager:v0.2.13. This will be the new default for new installs, but if you already have Rancher deployed and there is an "Upgrade Available" next to your "Network Services" stack, please upgrade to get your networking working.

For other people having issues, v1.2.0 only works with your docker bridge as docker0 [https://github.com/rancher/rancher/issues/6896] and docker being installed in `var/lib/docker` [https://github.com/rancher/rancher/issues/6897]. We are looking to make this more flexible in our next release.

**Comment by marvinwankersteen** [comment]
In my 2 server setup I use only docker0 as bridge interface. On the rancher-server I used /data/docker instead of /var/lib/docker. I moved the folder to /var/lib/docker, restartet docker, updated the network-services stack to 0.0.2 and the ipsec stack is running. The healthcheck and one other load-balancer is still initializing.

On the second server the folder was since install of docker (month ago) in /var/lib/docker also with bridge docker0 and the ipsec and healthcheck stack still bring the "Timeout gettings IP address".

**Comment by stormerider** [comment]
I'm running into this issue on a RancherOS set of nodes in AWS, which don't have the "rancher" command, so I'm not sure what I can do to provide the logs for that.

**Comment by firehist** [comment]
Thank you @deniseschannon .
My docker path was `/home/docker-data` use an existing partition. Your comment just fix my host connection (ipsec & healthcheck wasnt working). I'll follow #6897

**Comment by deniseschannon** [comment]
As always, everyone should make sure that the IP of their host is not the docker bridge IP but the actual IP of the host. Whenever you add a host that is also running the Rancher server container, it will pick up the docker bridge IP instead.

http://docs.rancher.com/rancher/v1.2/en/faqs/agents/#how-does-the-host-determine-ip-address-and-how-can-i-change-it-what-do-i-do-if-the-ip-of-my-host-has-changed-due-to-reboot

@marvinwankersteen Can you double check that your IPs are set correctly since you are using one of your 2 servers to run Rancher server and Rancher agent commands? 

@stormerider In AWS, have you made sure your security group is set up correctly? The `rancher` commands come from the [Rancher CLI](http://docs.rancher.com/rancher/v1.2/en/cli/). It is not related to the specific OS that is running on your hosts.

**Comment by stormerider** [comment]
Yeah, the SG is permissive enough ("All traffic" is allowed inside the SG for members of the SG), but I was using the internal IPs for the nodes instead of the public. I stopped the server, wiped all user containers (ie, anything from 'docker ps -a', but not touching anything from system-docker). Relaunched the server container (which uses an external RDS DB), deleted the hosts, and reconnected the hosts by recreating the agent containers, making sure to use the public IP instead. I did this host by host, letting the stacks spin up as well as they could before moving on to the next host. Now the healthchecker containers across all 3 nodes are still stuck in "Initializing" after several hours. The scheduler instance is also stuck similarly. And I see the hosts getting marked as disconnected frequently. I'm attaching the logs now that I installed the rancher cli tool (really unclear as to why that's missing from RancherOS, to be honest, that doesn't make sense to me...). I'm not sure if I'm dealing with the same issue as others at this point and I just got further down the process, or if this is a separate issue with 1.2.0 at this point:

[rancher-support-20161204-0609.tar.gz](https://github.com/rancher/rancher/files/629323/rancher-support-20161204-0609.tar.gz)

**Comment by hawkwang** [comment]
I got same problem as follows:
=================================================================
time="2016-12-04T15:10:17Z" level=info msg="Timeout waiting for ipsec to finish" 
time="2016-12-04T15:10:17Z" level=info msg="Timeout processing stack:1st6" 
time="2016-12-04T15:10:28Z" level=info msg="Finish Upgrade Stack Event Received" eventId=812100c0-c63d-421f-9380-03dd93e0db93 resourceId=1st6 
time="2016-12-04T15:10:32Z" level=info msg="Timeout waiting for ipsec to finish" 
time="2016-12-04T15:10:32Z" level=info msg="Timeout processing stack:1st6" 
time="2016-12-04T15:12:46Z" level=info msg="Finish Upgrade Stack Event Received" eventId=61738e29-ab80-4a35-9c25-1fe87d968048 resourceId=1st6 
time="2016-12-04T15:12:55Z" level=info msg="Timeout waiting for ipsec to finish" 
time="2016-12-04T15:12:55Z" level=info msg="Timeout processing stack:1st6" 
time="2016-12-04T15:13:02Z" level=info msg="Finish Upgrade Stack Event Received" eventId=61738e29-ab80-4a35-9c25-1fe87d968048 resourceId=1st6 
2016-12-04 15:13:11,115 ERROR [d8d5ce87-6888-46ed-8c31-24146f97f8cf:5748] [instance:195] [instance.start->(InstanceStart)] [] [ecutorService-6] [i.c.p.process.instance.InstanceStart] Failed to Waiting for dependencies for instance [195] 
time="2016-12-04T15:13:13Z" level=info msg="Timeout waiting for ipsec to finish" 
time="2016-12-04T15:13:13Z" level=info msg="Timeout processing stack:1st6" 
==================================================================

And the question is "can I increase timeout ?"

**Comment by samattridge** [comment]
@cjellick I'll try and sort out the logs this morning. Something that may help you diagnose the issue is that I just read issue #6897 regarding changing docker location from the default of /var/lib/docker. On the machine that is seeing the error I have symbolic linked that location so we can run it on another drive without changing too much configuration.  I'm assuming this could be the reason why I'm seeing this issue in particular but may not be the answer for others.

Unfortunately, we don't have much option other than to change the location of docker since we're using Azure and want to use a secondary SSD backed drive for the containers.

**Comment by marvinwankersteen** [comment]
@deniseschannon I double checked the ip address. On the rancher-server I added the host with the environment CATTLE_AGENT_IP and the right host ip address. Its not the bridge ip address.
The rancher-server server is running with all the container. The problem is the second server.
Maybe the problem is, that I have an openvpn to this server. So I have one public interface and one tun0 interface on the second server. I added the rancher-agent without the CATTLE_AGENT_IP. The server shows in Infrastructure --> Hosts the ip address of the tun0 interface. Perhaps this is wrong? Should it be the public address?
The second server reaches the rancher-server only over the openvpn.

**Comment by DomiStyle** [comment]
@deniseschannon 
> As always, everyone should make sure that the IP of their host is not the docker bridge IP but the actual IP of the host. Whenever you add a host that is also running the Rancher server container, it will pick up the docker bridge IP instead.

I checked multiple times already with raw Debian/Ubuntu installations. Which means the bridge is already called docker0, the IP range doesn't overlap and the docker directory is in /var/lib. I also tried manually passing the IP of the host along when adding the new host, which didn't work either. The correct IP address of the host is also shown in the host overview tab. Also, setting up a host on the same machine will fail too.

Is there any other debug output that actually gives insight in what exactly is failing? "Timeout getting IP address" isn't really telling me anything so I can't check any further.

**Comment by T-Grave** [comment]
When I add a host in a new environment it takes an enormous amount of time before all services turn green. It's constantly restarting everything (and I mean everything except the proxy and kubelet services) becuase of "Timeout getting IP address". After like 10 minutes everything seems to be running finally.

After that I tried adding a deployment from the UI, which depployed a pod that kept being destroyed / removed / restarted, without ever starting a container with my actual application.

Before all that, I upgraded from 1.1.4 and my existing Kubernetes environment got stuck at 7 of 9 services, with all etcd services stuck in initialization + a ton of red dots on my screen.

Also very annoying, once my enviroment is up and running it redirects me to the add your first stack page now and then, randomly, which is btw a broken page, the Add from catalog button is broken..

In my opinion, Rancher 1.2 should not have been released as a stable release yet. We are talking about a piece of software that people use to manage their infrastructure. I'm running a very simple environment with only 3 hosts and no more than 20 containers including k8s stack, yet it got completely wrecked by this update.

To anyone that hasn't upgraded yet, I would advice to wait for a more stable release.

**Comment by swartel-fa** [comment]
@samattridge A workaround I found to solve the issue related to /var/lib/docker is to use a mount point instead of a symlink.
For example: 
stop docker service
rm -f /var/lib/docker && mkdir -p /var/lib/docker
echo "</path/src>        /var/lib/docker         auto    rw,bind        0       0" >> /etc/fstab
mout -a
start docker service

Hope that helps !

**Comment by samattridge** [comment]
Thanks @swartel-fa for taking the time to reply.  Much appreciated!  I'll give that a go.

**Comment by mkurzeja** [comment]
@swartel-fa Thanks for the reply. It worked perfectly in my case

**Comment by swartel-fa** [comment]
@mkurzeja you welcome

**Comment by Gmentsik** [comment]
Just tried VXLAN - same issue

**Comment by galindro** [comment]
Same issue with me. I've tried with RancherOS v0.7.1, RancherOS v0.7.0 and Ubuntu 16.04. Docker versions 1.12.x

**Comment by DomiStyle** [comment]
I was able to find the problem and solve it. Since Rancher keeps reporting it can't get some IP address I started sniffing all DNS queries on my DNS server and found the following:

    Dec 9 19:13:46 	unbound 	51000:0 	info: 192.168.1.xxx rancher-metadata.example.org. A IN
    Dec 9 19:13:42 	unbound 	51000:0 	info: 192.168.1.xxx rancher-metadata.example.org. A IN
(where example.org is my internal domain.)

Obviously this isn't supposed to happen, *.example.org will always resolve to my public IP unless it is overwritten by the local DNS server. I never bothered changing this since I didn't think it would cause any issues.

Anyways, *.example.org will not resolve on my internal network anymore unless it is defined internally. This fixed the issue. Rancher will still issue the useless DNS request but won't get a valid response anymore.

I still think this probably should be fixed? I doesn't look like this DNS lookup should reach the network.

**Comment by deniseschannon** [comment]
@stormerider I finally had a chance to look at your logs. Your issue seems different as it looks as if your `go-machine-service` is flapping in your server. Could you open a new issue for that? 

@marvinwankersteen The IP address of the the server running rancher/agent should use the public IP address unless the rancher/server server can access the `tun0` IP. 

@hawkwang There is no way to increase the timeout. @DomiStyle There is nothing else distinctive evidence except for providing us logs. 

@hawkwang  If you use the [Rancher CLI](http://docs.rancher.com/rancher/v1.2/en/cli/), you can provide these logs. Note: this will share the IP address of your hosts. 

```
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
``` 

@DomiStyle I'll forward your latest update to our team to see what's going on. 


@T-Grave I've seen the k8s issue of the UI before. If you manually refresh the page, it'll start working. I haven't been able to pinpoint what are the exact steps to get in that loop where you can't access the catalog.

**Comment by tnh** [comment]
are you enabling ICMP between servers? maybe thats it?

**Comment by IrisDande** [comment]
@swartel-fa 
i dont know
echo "</path/src>        /var/lib/docker         auto    rw,bind        0       0" >> /etc/fstab
what is </path/src>

**Comment by VincenzoFerme** [comment]
@IrisDande the path to your Docker working directory, e.g., `/data/docker`.

**Comment by IrisDande** [comment]
@VincenzoFerme
But what is correct path if docker is installed from repo by default?

**Comment by VincenzoFerme** [comment]
I that is the case, then I think this workaround does not apply to you, because the directory should already be the default one: `/var/lib/docker`

**Comment by IrisDande** [comment]
@VincenzoFerme  did you have any solution in that case? i can't install old version of rancher (old version did meet this error)

**Comment by VincenzoFerme** [comment]
Can you report all the information suggested on the following issue template: https://github.com/rancher/rancher/blob/master/.github/ISSUE_TEMPLATE.md?

**Comment by IrisDande** [comment]
@VincenzoFerme  Sorry but my issue  have same with thread ( no different ) so i dont want to add a duplicate issue  :(

**Comment by VincenzoFerme** [comment]
I mean, you can report the requested information in this issue, if they are different than the one discussed in the first comment of this issue: https://github.com/rancher/rancher/issues/6858#issue-192896871

**Comment by IrisDande** [comment]
Rancher Version: 1.2.0
Docker Version: 1.12.3, build 6b644ec
OS and where are the hosts located? (cloud, bare metal, etc): Debian 8 on VMware ESXI
Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB): single node rancher, internal DB
Environment Type: (Cattle/Kubernetes/Swarm/Mesos): Cattle 
Steps to Reproduce:
- Install docker on debian 8 (or ubuntu 16.04)
- Install Rancher/server on this server
- Install Rancher/Agent:v1.1.0 on this server

Results: Expected state running but got error: Timeout getting IP address

Expected:

**Comment by kkvilekval** [comment]
This issue kept several of our bare-metal Ubuntu 16.04 servers off network for over a week. 

A work-around for me was:

1.  Remove any reference to rancher (kill and remove containers), and umount /var/lib/rancher/volumes
2.  rm -rf /var/lib/rancher 
3.  **apt-get purge docker-engine** 
4.  **rm -rf /var/lib/docker** 
5.  apt-get install docker-engine  
6.  install rancher-agent (however you choose).

.

**Comment by marrotte** [comment]
Same here.  Fresh install ESXi 6.0, Ubuntu 16.04, latest everything else following the Quick Start Guide.  

What sort of attention is this getting at Rancher?

**Comment by deniseschannon** [comment]
@marrotte There are multiple reasons why this error could have occurred as we have outlined in this issue. 

If you have boot2docker hosts, you need to make sure you upgrade to the latest network services. 

There are known issues with if your docker is installed in a different location or if your docker bridge is not at docker0. 

We are working on 1.2.1 that will address the above issues. For all other cases, we've asked for users to use the Rancher CLI to provide us with the following information so we can help debug:

```
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
```

Note: This would show the IP of your hosts.

**Comment by kkvilekval** [comment]
After retrying the apt-get purge docker-engine  solution, it is failing again.

Where do we send the logs?

**Comment by deniseschannon** [comment]
@kkvilekval You can send them to darren@rancher.com and/or me (denise@rancher.com)

**Comment by proligde** [comment]
We have the same problem here after upgrading to Rancher 1.2.0 with a rancher server on a current rancher os release and several clients running ubuntu 16.04. No boot2docker, no non-standard docker-root. We deleted all docker containers on the agent and switched to 1.2.1-rc2 (using agent version 1.1.1-rc1) but it didn't help so far. We will downgrade to 1.1 now.

**Comment by justsml** [comment]
Me too, have similar setup to  @IrisDande .

Thanks @deniseschannon - I'll send you and Darren my log output

**Comment by garjunan** [comment]
+1 

I am also experiencing same error. 'Timeout getting IP address' error on healthcheck and ipsec  containers and it is keep restarting. But my own stacks running fine and i am not sure about the impacts due to above failed containers. . So I have reverted the upgrade to previous version(1.1.4).

Can we expect the fix quickly for this issue?

**Comment by proligde** [comment]
I forgot to add: Our apllication stacks were looking like they'd be running fine but then we saw that the containers couldn't connect to each other: Even the ones running within the same stack on the same host. And this was **not** a DNS issue - we couldn't ping the other containers even using their IP addresses. Unfortunately I couldn't find out whether this was a consequence or the (same) reason of the degraded infrastructure stack.

**Comment by jwhitcraft** [comment]
I'm seeing this as well,

Ubuntu 12.04 with Docker 1.12.3 and on AWS.

All the hosts can talk to each other just fine, just sometimes the it can't get the IP.  I've built in re-tires to our deploy scripts but it would be nice to not have to do that.  Also

@deniseschannon  or @ibuildthecloud  Can you provide me what logs you need and I can not tell from from all the comments.

**Comment by deniseschannon** [comment]
@jwhitcraft 

```
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
```

**Comment by ckreutz** [comment]
Just want to add I have the same issues and that **the latest version 1.2.1 version does not solve it for me**. I tried it with various servers on Digital Ocean (Ubuntu, CoreOS etc.) and tried the mentioned hints above. (Sent my logs too). Did not work for me and I had no choice but to downgrade to 1.4.1. 
1.2 is really not a resilient/stable version for me.

**Comment by garjunan** [comment]
Hi,

healthcheck and ipsec containers came up for me after upgrading it to 1.2.1. But none of my services are functional. Name resolution is working but getting connection time out from one service to other.

For ex: service1 connecting to Mongo service on same stack using service link/alias.  I am able resolve the hostname of mongo but connection is timing out. Mongo is up and i am able to connect directly. 

**WARNING: Server seen down: DB-MONGO-1/198.105.244.11:27017 - java.io.IOException - message: connect timed out**

I had to revert back to 1.1.4 without any choice.

**Comment by ddelnano** [comment]
I created a brand new Rancher 1.2 and 1.2.1 setup and am unable to get it the infrastructure services deployed.   I emailed my logs.  Its a shame that 1.2 doesn't seem to work easily out of the box.  Just to detail what I am using

**Rancher Version:**
v1.2.0-1 and v1.2.1

**Docker Version:**
1.12.4

**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 16.04 AWS

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node external db and HA Rancher

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Create brand new HA Rancher cluster.  Add a host to an environment.

**Results:** 
All infrastructure services except metadata and network manager fail to start.

**Comment by dax0** [comment]
There are an incredible amount of issues with Rancher 1.2.1 we've burned a huge amount of time dealing with this.  It is clearly not ready for production.  You guys need to revert the :latest tag to something stable.  You are screwing more and more people over.

**Comment by tnh** [comment]
We failed in a rouge upgrade to Rancher 1.2.x (We didnt pin the version of our 'rancher-server' and an EC2 instance rebooted and force upgraded us). 

Our topology is
```
[Rancher server (HA set with RDS DB)]  <--->   [N number of instances that add specific labels that register with Rancher Server]     
```

We are hosted in AWS and we have tight security groups between our instances and the rancher server. We block ICMP by default, but enable HTTP/ws to Rancher Server and UDP ports for health checks in Rancher 1.1
 
I have a feeling that the new upgrade and new network containers that are launched as a result need ICMP. I wasnt able to confirm as we had to abandon trying to force upgrade.

**Comment by proligde** [comment]
@tnh that's interesting as it matches our setup exactly, where it failed as well: rancher server on aws, strict firewalling with no icmp, external DB on rds. And we use an ELB as our rancher endpoint, forwarding the TCP traffic to the rancher servers using an ELB proxy policy to make websockets work correctly. Hope this helps others or the rancher dev team to find parallels causing the issue. Or is the reason known already at the rancher team?

**Comment by garjunan** [comment]
Hi Team,

Any update? Is 1.2.2 has fix for this issue?

**Comment by giovannicandido** [comment]
Hi, I found a solution to this problem that works on ubuntu server, and desktop as well.

The cause is that ubuntu ships by default using dns server as local address 127.0.0.1 and is by design. Problem is docker containers can't lookup in 127.0.0.1

To diagnose you can do the follow procedure:

See your **/etc/resolv.conf** file
Perform a test:

```
docker run --it ubuntu bash
apt update
apt install dnsutils
# This will not respond
dig @127.0.0.1 your.hostname.com
```
Note: ping WILL work fine, and could trick you in thinking that name resolution is working. Dig is a proper way of doing that. 

You can also use rancher cli to get a hint of the problem:

```
mkdir -p support
rancher hosts -a > support/hosts
rancher logs --tail=-1 ipsec/ipsec > support/ipsec 2>&1
rancher logs --tail=-1 network-services/metadata > support/metadata 2>&1
rancher logs --tail=-1 network-services/network-manager > support/network-manager 2>&1
```

The error will be like `network-services-metadata-1 | time="2016-12-26T23:58:21Z" level=fatal msg="Failed to subscribeGet http://xx.xxx.xxx:xxx/v2-beta: dial tcp: lookup xxx.xxx.xxx.xxx on 127.0.0.1:53: read udp 127.0.0.1:44285->127.0.0.1:53: read: connection refused" 
`


There are two solutions:
1 - Configure ubuntu to use other nameserver like google public dns (8.8.8.8, 8.8.4.4). I try this one, and is by far too complicated for a simple change, as a said, ubuntu use that by design.
2 - Change docker dns server. This worked fine for me. You will edit or create the file **/etc/docker/daemon.json** and put the line:


```
{
  "dns": ["8.8.8.8", "8.8.4.4"]
}
```

Stop the containers and Restart the daemon:

```
docker stop $(docker ps -q)
docker stop $(docker ps -q) # yes twice :-) rancher will try do restart your dying containers
systemctl restart docker
```

Oh boy, after days in this problem, I will drink a could beer :-)

Thank you @deniseschannon, your procedure of using rancher cli to generate logs was what give me the hint necessary to find the cause.

Question: Is there some problem in using a external DNS, with rancher ?

And finally, for the ones who want the solution 1, I let some links to get started (none of the solutions worked for me :-( 

http://askubuntu.com/questions/143819/how-do-i-configure-my-static-dns-in-interfaces
http://askubuntu.com/questions/327532/why-would-127-0-0-1-in-resolv-conf-cause-problems-in-dns-resolution
http://askubuntu.com/questions/627899/nameserver-127-0-1-1-in-resolv-conf-wont-go-away

**Comment by jwhitcraft** [comment]
@giovannicandido 

Thanks for the solution but it doesn't fix the problem i'm seeing.

After bringing a new host online, it starts the network-services-network-metadata but this is what i get in the logs

```
time="2016-12-27T03:01:48Z" level=info msg="Waiting for metadata"
Creating metadata client: Get http://rancher-metadata/2016-07-29/version: dial tcp: lookup rancher-metadata on 169.254.169.250:53: read udp 10.42.0.1:35338->169.254.169.250:53: i/o timeout
time="2016-12-27T03:04:01Z" level=info msg="Waiting for metadata"
Creating metadata client: Get http://rancher-metadata/2016-07-29/version: dial tcp: lookup rancher-metadata on 169.254.169.250:53: read udp 10.42.0.1:34519->169.254.169.250:53: i/o timeout
time="2016-12-27T03:06:13Z" level=info msg="Waiting for metadata"
Creating metadata client: Get http://rancher-metadata/2016-07-29/version: dial tcp: lookup rancher-metadata on 169.254.169.250:53: read udp 10.42.0.1:56142->169.254.169.250:53: i/o timeout
```

And even on my second host i bring up, i still get the the ip timeout errors.  I'm about to scrap this whole 1.2.2. environment and go back to 1.1.4 till they can get this figured out.

**Comment by ibuildthecloud** [comment]
@jwhitcraft Can you just double check the output of `ip route` on the host and additionally, are you running with the default `docker0` bridge?  I'm looking through the last set of info you sent me now.

**Comment by jwhitcraft** [comment]
@ibuildthecloud, sadly, I've already deleted that environment.  I'm in the process of testing out an environment with a vxlan instead of ipsec and so far it's looking good.  I've added 4 hosts now and they've all come up the first time with no issues 🤞 that it keeps up.

If i run into it again i'll let you know.

**Comment by ibuildthecloud** [comment]
@jwhitcraft There's no fundamental difference between IPsec and vxlan in how it is managed.  So if you repeatably had issue with IPsec, I would not expect VXLAN to be better.  

You had previously sent a zip with two servers.  Do you know if the resolv.conf in the zip was from the  metadata-dns container?  The container running the rancher-dns process?

**Comment by jwhitcraft** [comment]
@ibuildthecloud,

Yes they came from the sidekick containers which are the metadata-dns servers.

**Comment by ibuildthecloud** [comment]
@jwhitcraft Looking at your answers.json I can see that rancher-dns is in fact recursing to itself "169.254.169.250".  That is basically what is causing your issue.  Now why it's configured as such, no clue.  Let me come up with some theories.

**Comment by jwhitcraft** [comment]
@ibuildthecloud, the hosts are on Ubuntu 16.04 and 14.04 fully patched with docker 1.12.3 or newer.  Tonight i upgraded them all to 1.12.5 before trying the vxlan tests.

**Comment by ibuildthecloud** [comment]
@jwhitcraft You don't by chance have either `169.254.169.250` set in the host `/etc/resolv.conf` or have `--dns 169.254.169.250` set on the Docker daemon?

**Comment by jwhitcraft** [comment]
My resolv.conf points to my companies internal dns servers and the docker daemon doesn't anything extra set on it, just a stock `apt-get install docker-engine` was run.

**Comment by ibuildthecloud** [comment]
@jwhitcraft Do you have any broken hosts left?  We run rancher-dns as `--net container:metadata` and then `metadata` runs `--net bridge`.  For whatever reason the resolv.conf in `metadata` is getting `nameserver 169.254.169.250` set.  If you happen to have a broken host if you do a `docker inspect` on the `r-network-services-metadata-XXXXX` you should see the HostConfig DNS set as follows:

```
"Dns": null,
"DnsOptions": null,
"DnsSearch": [
    "network-services.rancher.internal",
    "metadata.network-services.rancher.internal"
],
```

So basically we tell Docker to just use the default DNS server, which would be what is in `/etc/resolv.conf` on the host.

So I only have two theories right now.

1) Docker was configured as 169.254.169.250 as the DNS:  I'm guess this is no because you said it was a default setup
2) Some how Rancher managed to edit the hosts /etc/resolv.conf.  The only way I could imagine that is if maybe `--net host -l io.rancher.container.dns=true` has a bad behavior in docker 1.12 and we end up actually changing the host DNS.  I'm testing this now.

**Comment by ibuildthecloud** [comment]
@jwhitcraft It's got to be option two.  Do you ever run a container with `-v /etc/resolv.conf:/etc/resolv.conf` That would cause it.

**Comment by jwhitcraft** [comment]
Naw, i'll email you the compose files that we use, only the image name and variables change in it.  We don't mount anything to the hosts.

**Comment by ibuildthecloud** [comment]
@jwhitcraft Thanks, I also realize that while `-v /etc/resolv.conf:/etc/resolv.conf` has a really bad behavior in Docker 1.12, it won't exactly cause the issue you are seeing.

**Comment by proligde** [comment]
I can confirm that adding external DNS as @giovannicandido suggested, fixed the issue in my case. Thanks a lot, man!

> 2 - Change docker dns server. This worked fine for me. You will edit or create the file /etc/docker/daemon.json and put the line:
> {
  "dns": ["8.8.8.8", "8.8.4.4"]
}

**Comment by ibuildthecloud** [comment]
## Update

A lot of people are experiencing issues with Rancher w/ a setup where a caching DNS resolver is used on the host. @giovannicandido pointed out is the case with Ubuntu.  The root cause of this is basically what I believe is a bug in Docker in which Docker does not set the DNS nameserver properly when only the `DnsSearch` option is used.  For futher reference I've created this bug in Docker https://github.com/docker/docker/issues/29815

In Rancher 1.3 (due out this week) we are putting in work around to deal with this behavior in Docker.  For Rancher 1.2.x the best approach is to set `--dns` on the `docker daemon` command or set the below in `/etc/docker/daemon.json`:

```
{
  "dns": ["8.8.8.8", "8.8.4.4"]
}
```

Sorry for the troubles here.  We do test quite extensively on Ubuntu but not with a configuration that has a local dnsmasq resolver and as such we didn't catch this one.

**Comment by justsml** [comment]
Thx again @ibuildthecloud

**Comment by marvinwankersteen** [comment]
I have did some debugging for the dns-issue.
On the last host I have problems I get this error-message:
> Creating metadata client: Get http://rancher-metadata/2016-07-29/version: dial tcp: lookup rancher-metadata on 169.254.169.250:53: read udp 172.17.0.1:42662->169.254.169.250:53: i/o timeout

If I try to do a `ping rancher-metadata.rancher.internal` in the container `network-services-network-manager-4` the dns-name is not resolved.
So I edited the /etc/hosts and added `169.254.169.250 rancher-metadata.rancher.internal`.

Now I get the ip IN the container. Then I tried to do the `curl http://rancher-metadata/2016-07-29/version`.

But this is not possible. It cant reach the IP 169.254.169.250 with HTTP Port. I have no idea what exactly is the problem.
I have some custom iptables rules. But I set for every chain the policy to ACCEPT and also inputed an -j ACCEPT on position 1 in INPUT, FORWARD and OUTPUT.

**Comment by balzaczyy** [comment]
In my case, my network-services/metadata got the correct resolv.conf settings. But network-manager got the "incorrect" resolv.conf. It doesn't make sense.

I tried the --dns and daemon.json but they don't work for me.

**Comment by balzaczyy** [comment]
Turns out that I added proxy settings when adding the host. Removing the proxy settings resolved my issue.

**Comment by aemneina** [comment]
@marvinwankersteen on your agent hosts do you have a dns caching service running, like nscd?

**Comment by JaviSabalete** [comment]
Several problems here too.

__Only with Ubuntu 16.04 __. Ubuntu 14.04 and Debian 8.5 works correctly.

Fixed one ubuntu 16 with these steps:
-removing host and stacks
-create new environment
-service docker stop
-delete dns=dnsmasq from /etc/NetworkManager/NetworkManager.conf
-service network-manager restart
-service docker start
-adding host to the new environment

But... I can do this on 20 machines. I'm restoring 1.1.4 atm.

btw... same issue upgrading with 1.2.1, 1.2.2 and 1.3.0

**Comment by marvinwankersteen** [comment]
@aemneina No, nothing like that.
But I found the problem on my server. The problem was the iptables-rules. The rules were to strict. I just allowed traffic to specific ports in the FORWARD-chain. But not between the containers.

The node is reachable from external and I want to set some iptables-rules. But it where to strict. So I edited my rules and now it works. Which rule exactly was the problem I have no idea.

@Rancher-Team: It would be nice if you can tell us which rules MUST exist that the containers can communicate among each other. Or add them automatic.

**Comment by mattdodge** [comment]
After upgrading to 1.3.0 I'm still seeing the Timeout Getting IP Address on one of the two Rancher environments I had set up previously. Interestingly, one of the environments works, but the other one won't let me start any stacks/services because of that error. Even the Rancher Infrastructure stacks (like scheduler, healthcheck, etc) aren't able to be started or upgraded.

When I `docker inspect` the `network-services-metadata-###` containers in both environments the DNS records all look the same:
```
"dnsSearch": [
  "network-services.rancher.internal",
  "metadata.network-services.rancher.internal"
]
```

The "broken" environment consists of two hosts, one running RancherOS 0.6.1 (docker version 1.12.1) and the other runs Ubuntu 14.04 (docker version 1.10.3). No changes to the system resolv.conf or DNS settings. The timeout error happens on both hosts. Here's a likely relevant line from the `network-manager-X` logs from a host when trying to bring up a stack.

```
network-services-network-manager-1 | time="2017-01-08T18:13:44Z" level=error msg="Failed to evaluate network state for 6f43c8282c1e8934b5d78e88703e591cf5c5a2c4d48a9c4f02db003eddc82137: Bringing up networking: No IP address found"
network-services-network-manager-1 | time="2017-01-08T18:13:44Z" level=error msg="Error processing event &docker.APIEvents{Action:\"start\", Type:\"container\", Actor:docker.APIActor{ID:\"6f43c8282c1e8934b5d78e88703e591cf5c5a2c4d48a9c4f02db003eddc82137\", Attributes:map[string]string{\"image\":\"rancher/scheduler:v0.5.0\", \"io.rancher.container.agent_id\":\"34223\", \"io.rancher.container.name\":\"scheduler-scheduler-1\", \"io.rancher.service.hash\":\"fac0a074c6689097fbbfb12e410bab30c3d07941\", \"io.rancher.cni.network\":\"ipsec\", \"io.rancher.project.name\":\"scheduler\", \"io.rancher.container.ip\":\"10.42.99.87/16\", \"io.rancher.service.deployment.unit\":\"713893f6-c035-4f45-8c81-21d62299cfb4\", \"io.rancher.service.launch.config\":\"io.rancher.service.primary.launch.config\", \"io.rancher.stack_service.name\":\"scheduler/scheduler\", \"io.rancher.cni.wait\":\"true\", \"io.rancher.container.agent_service.scheduling\":\"true\", \"io.rancher.container.create_agent\":\"true\", \"io.rancher.container.uuid\":\"20449261-40dc-4acb-9fb8-13785771bf1f\", \"io.rancher.project_service.name\":\"scheduler/scheduler\", \"io.rancher.stack.name\":\"scheduler\", \"name\":\"r-scheduler-scheduler-1-20449261\"}}, Status:\"start\", ID:\"6f43c8282c1e8934b5d78e88703e591cf5c5a2c4d48a9c4f02db003eddc82137\", From:\"rancher/scheduler:v0.5.0\", Time:1483899103, TimeNano:1483899103900821540}. Error: Bringing up networking: No IP address found"
network-services-network-manager-1 | time="2017-01-08T18:13:46Z" level=info msg="Evaluating state from retry" cid=6f43c8282c1e8934b5d78e88703e591cf5c5a2c4d48a9c4f02db003eddc82137
```

This environment worked with Rancher 1.1.4. Anything I can test? I figure I may be in a unique and potentially interesting situation by having one environment that works and one that does not.

**Comment by JaviSabalete** [comment]
I tested with clean db (because #7245) and no errors happens

**Comment by jwhitcraft** [comment]
@JaviSabalete That is what I had to do, my old rancher install was so unreliable that I just started over on 1.3.0 and it's been humming along ever since.  I send my database dump to them to see if they could find anything.

**Comment by deniseschannon** [comment]
@mattdodge try stopping the scheduler and see if your network-services reconcile before starting the scheduler again. This may be related to

https://github.com/rancher/rancher/issues/7375

**Comment by deniseschannon** [comment]
There have been many releases of our infrastructure services to help address the issues relating to timeouts. I will be closing this issue for now so that it's easier to track any new issues with the same message.

As we've tried to state before, there have been many reasons why the stacks might get into this state and we've tried to isolate the issues as they've come in and push out updates as quickly as possible. 

Please try again with the latest version v1.3.0 and the default infra stacks for that version. If you're upgrading, please read the Release Notes regarding upgrading.

https://github.com/rancher/rancher/releases/tag/v1.3.0

**Comment by mattdodge** [comment]
Stopping the scheduler didn't resolve the issue for me. I even upgraded network-services while it was stopped and then restarted the scheduler. At that point the scheduler wouldn't even start, it got stuck in "Initializing" state. There are plenty of other issues relating to this so no need to go into it here.

It looks as though I should just start over from scratch, something in my set up is clearly hosed. I'd been meaning to clean up our AWS and Rancher environments anyways, I guess this is as good an excuse as any to make it happen.

**Comment by proligde** [comment]
After still having the 'Timeout getting IP address' issue on some hosts on Rancher 1.3.0 I found another workaround I'd like to share. The /etc/resolv.conf inside docker not only inherits the host system's dns settings but also its search host(s) which as a result can lead to the same problem.

I was able to fix that by overwriting the dns-search hosts adding a new line to the suggested entry in /etc/docker/daemon.json (replace example.org with a proper search domain in your network)

`
{
  "dns": ["8.8.8.8", "8.8.4.4"],
  "dns-search": ["example.org"]
}
`

Please note that dns-search may neither be empty (the hosts' search domains would still be inherited) nor contain an empty string (your docker service wouldn't start anymore). Hope this helps some of you guys out there.

**Comment by dren-dk** [comment]
@DomiStyle I've been having the exact same problem with rancher 1.3.0 and 1.3.3, I finally tried your approach and started looking at DNS traffic with tcp dump and my failing host did indeed look up rancher-metadata in all the search domains of the host, until it got to the public company domain which has a * record that resolves to an external website.

My solution was to remove the internet-only domain from /etc/resolv.conf and that caused ipsec to start, now I just have healthcheck stuck in initializing.

I really wish the internal names looked up by services such as ipsec will be changed to use the fqdn ending in .internal. so we can avoid both spamming the external DNS with useless requests and this problem in particular.

**Comment by nextrevision** [comment]
I've been seeing the "Timeout getting IP address" issue off and on, some containers schedule, some don't. After looking at the `network-services/metadata` logs, I saw that about half of them had the following entries near the end (IPs/hostnames changed):

```
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Loading answers"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Loaded answers"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Applied https://rancher.acme.com/v1/configcontent/metadata-answers?version=15755-8b84aebc80dd7a28cb0d0e082d379dd7"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Downloaded in 230.423534ms"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Loading answers"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Loaded answers"
network-services-metadata-9     | time="2017-01-27T14:08:56Z" level=info msg="Applied https://rancher.acme.com/v1/configcontent/metadata-answers?version=15755-8b84aebc80dd7a28cb0d0e082d379dd7"
network-services-metadata-9     | time="2017-01-27T15:33:03Z" level=info msg="Hit websocket pong timeout. Last websocket ping received at 2017-01-27 15:32:50.015981928 +0000 UTC. Closing connection."
network-services-metadata-9     | time="2017-01-27T15:33:03Z" level=warning msg="websocket closed: websocket: close sent"
network-services-metadata-9     | time="2017-01-27T15:33:04Z" level=warning msg="websocket closed: websocket: close sent"
network-services-metadata-9     | time="2017-01-27T15:33:04Z" level=info msg="Initializing event router" workerCount=3
network-services-metadata-9     | time="2017-01-27T15:33:05Z" level=warning msg="websocket closed: websocket: close sent"
network-services-metadata-9     | time="2017-01-27T15:33:34Z" level=error msg="Failed to download and reload metadata: Get https://rancher.acme.com/v1/configcontent/metadata-answers: dial tcp 10.x.x.x:443: i/o timeout"
```

After the websocket closes, it doesn't appear to reopen on it's own (it's less alarming to me why the websocket closed). I restarted all the metadata containers that had closed the connection and the containers I saw receiving the "Timeout getting IP address" error finally started.

**Comment by pschrammel** [comment]
what's disturbing me is the download time:
ime="2017-01-31T18:38:06Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=127595-8b84aebc80dd7a28cb0d0e082d379dd7" 
time="2017-01-31T18:38:25Z" level=info msg="Downloaded in 18.454818263s" 

it takes up to 24 secs to download, I have 27 servers running the rancher DB currently gets up to 3000 req/s and has a 50% load (RDS aurora r3.large - that's crazy)

**Comment by pschrammel** [comment]
Theory:
it takes around one minute download and apply the metadata. So in some cases it can take over 2 minutes until an IP is visible. This runs into the 120 sec timeout of the agent.

Some logs:
`
31.1.2017 20:18:40time="2017-01-31T19:18:40Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44878-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:19:00time="2017-01-31T19:19:00Z" level=info msg="Downloaded in 19.787896964s"
31.1.2017 20:19:20time="2017-01-31T19:19:20Z" level=info msg="Loading answers"
31.1.2017 20:19:21time="2017-01-31T19:19:21Z" level=info msg="Loaded answers"
31.1.2017 20:19:21time="2017-01-31T19:19:21Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44922-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:19:45time="2017-01-31T19:19:45Z" level=info msg="Downloaded in 23.568940433s"
31.1.2017 20:20:02time="2017-01-31T19:20:02Z" level=info msg="Loading answers"
31.1.2017 20:20:03time="2017-01-31T19:20:03Z" level=info msg="Loaded answers"
31.1.2017 20:20:03time="2017-01-31T19:20:03Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44958-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:20:22time="2017-01-31T19:20:22Z" level=info msg="Downloaded in 18.835192201s"
31.1.2017 20:20:42time="2017-01-31T19:20:42Z" level=info msg="Loading answers"
`

**Comment by miv** [comment]
This has happened to me once in two days for last couple of weeks. Restart of the node helps, but i found out that restart of network-manager and dns helps also

**Comment by proligde** [comment]
@pschrammel @miv I think we're going off topic here but just in case it helps you: We had similrar issues with high load/traffic and found out that there were stacks containing flapping services that were stopping/starting all the time. Although that were only a few containers that produced a majority of the load and traffic we were experiencing. After cleaning that up the load on our rancher server is minimal.

**Comment by raphink** [comment]
We're still seeing this issue with 1.3.4.

**Comment by pschrammel** [comment]
Theory:
it takes around one minute download and apply the metadata. So in some cases it can take over 2 minutes until an IP is visible. This runs into the 120 sec timeout of the agent.

Some logs:
`
31.1.2017 20:18:40time="2017-01-31T19:18:40Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44878-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:19:00time="2017-01-31T19:19:00Z" level=info msg="Downloaded in 19.787896964s"
31.1.2017 20:19:20time="2017-01-31T19:19:20Z" level=info msg="Loading answers"
31.1.2017 20:19:21time="2017-01-31T19:19:21Z" level=info msg="Loaded answers"
31.1.2017 20:19:21time="2017-01-31T19:19:21Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44922-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:19:45time="2017-01-31T19:19:45Z" level=info msg="Downloaded in 23.568940433s"
31.1.2017 20:20:02time="2017-01-31T19:20:02Z" level=info msg="Loading answers"
31.1.2017 20:20:03time="2017-01-31T19:20:03Z" level=info msg="Loaded answers"
31.1.2017 20:20:03time="2017-01-31T19:20:03Z" level=info msg="Applied https://xxx/v1/configcontent/metadata-answers?version=44958-8b84aebc80dd7a28cb0d0e082d379dd7"
31.1.2017 20:20:22time="2017-01-31T19:20:22Z" level=info msg="Downloaded in 18.835192201s"
31.1.2017 20:20:42time="2017-01-31T19:20:42Z" level=info msg="Loading answers"
`

**Comment by Antauri** [comment]
One of our sys-admins tracked this down to wildcard DNS configurations on our servers. Maybe that helps someone here.

**Comment by gcristofol** [comment]
Still happens with rancher/server: 1.5.2 and rancher/agent: 1.2.1. 

It is something related to the environment (as in cattle environment) creating a new environment and adding hosts to it works just fine, and there's more, once the host has been added, it can be deleted from the fresh environment and it works just fine in the 'broken' environment.

I reproduced quite a few times, getting snapshots of the database in the process

**Comment by yevheniidehtiar** [comment]
What the f...g problem. It still happens every time after system upgrade or rancher upgrade. Sometimes it happens after reboot host. I use rancher from beta version... it was 0.6 and it is best open-source UI GUI for docker, kubernetes and other services for DevOps. But why you, rancher team, can not to resolve this problem with your Cattle Stack. I don't want to remove and recreating on all hosts new environment every time, after my home-server rebooting. I will have good sleeping and don't think about possible situations or pray to Digital God for success starting services and containers after electricity interruptions. 
Can you research this problem and describe all possible variants of guides how to issue this global problem! I hope that you'l decide to do it right.

**Comment by marvinwankersteen** [comment]
Please check if you hit this problem:
https://github.com/rancher/rancher/issues/9367#issuecomment-315893457

I had the same problem and could solved it.

**Comment by alach1** [comment]
Also check the https://github.com/rancher/rancher/issues/8383#issuecomment-322617101, one of our hosts had docker bridge on IP different from  '172.17.0.1', and after forcing it to  '172.17.0.1' 
`
/etc/docker/daemon.json:
{
  "bip": "172.17.0.1/16"
}
`

all our stack works like a charm, i can bounce any hosts and it goes back to normal. Though whole night spent on it.
